storage/src/vespa/storage/frameworkimpl/status/statuswebserver.cpp

```diff
@@ -112,45 +72,47 @@ void StatusWebServer::configure(std::unique_ptr<vespa::config::content::core::St
 }
 
 StatusWebServer::WebServer::WebServer(StatusWebServer& status, uint16_t port)
-    : Fast_HTTPServer(port, NULL, 100, false, 128*1024, 10),
-      _status(status),
-      _serverSpec(vespalib::make_string("%s:%d", vespalib::HostName::get().c_str(), port))
+    : _status(status),
+      _server(vespalib::Portal::create(vespalib::CryptoEngine::get_default(), port)),
+      _executor(1, 256 * 1024),
+      _root(_server->bind("/", *this))
 {
 }
 
+StatusWebServer::WebServer::~WebServer()
+{
+    _root.reset();
+    _executor.shutdown().sync();    
+}
 
 namespace {
-    /** Utility class for printing HTTP errors. */
-    struct HttpErrorWriter {
-        std::ostream& _out;
 
-        HttpErrorWriter(std::ostream& out, vespalib::stringref error)
-            : _out(out)
-        {
-            _out << "HTTP/1.1 " << error << "\r\n"
-                    "Connection: Close\r\n"
-                    "Content-type: text/html\r\n\r\n"
-                    "<html><head><title>" << error << "</title></head>\r\n"
-                    "<body><h1>" << error << "</h1>\r\n"
-                    "<p>";
-        }
+struct HandleGetTask : vespalib::Executor::Task {
+    vespalib::Portal::GetRequest request;
+    std::function<void(vespalib::Portal::GetRequest)> fun;
+    HandleGetTask(vespalib::Portal::GetRequest request_in,
+                  std::function<void(vespalib::Portal::GetRequest)> fun_in)
+        : request(std::move(request_in)), fun(fun_in) {}
```

Comment:
Since `fun_in` is passed by value, consider constructing `fun` by move

Reply:
done